### DISCUSSION TITLE
Help Desk 🔧🪛

### DISCUSSION THREAD

**Discussion by waju-ire** [wagtail/outreachy#30] [General]
Hey guys, to not overload the project discussions, I think we can have this place to talk about issues we may face outside the projects. Issues like GitHub familiarity for those that are new to it and any and every issue relevant to help our contributions.

**Comment by waju-ire** [comment]
Hi @triumphant-fortune  I saw your issue with the markdown table. This link can be of help to you:
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables.

Also, this link can help with basic writing and formatting: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links

There's this YT video I found on getting familiar with GitHub as a starter:
 https://www.youtube.com/watch?v=mJ-qvsxPHpY&t=599s

**Reply by praiztech** [reply]
Great idea @waju-ire! Thanks for the resources.

**Reply by Temiwaji** [reply]
Thank you very much @waju-ire

**Reply by triumphant-fortune** [reply]
@waju-ire  Thanks so much

**Comment by waju-ire** [comment]
And here's that playlist to refresh/introduce you to accessibility: https://www.youtube.com/playlist?list=PLWSYD-KxdDxvxYQ2UHixf6vvlbhvUzXNB
 Enjoy😊

**Reply by Temiwaji** [reply]
Needed... Thank you @waju-ire

**Comment by thibaudcolas** [comment]
Great idea @waju-ire 😛

**Comment by Fatima-Bint** [comment]
Hello guys,
Here's a simplifies guide to contributing directly on the GitHub interface. It's a public gist so anyone can also contribute to make it better.
https://gist.github.com/Fatima-Bint/2b9acb0316972065825664224013f1b2

**Reply by Temiwaji** [reply]
Thank you @Fatima-Bint

**Reply by thibaudcolas** [reply]
great stuff!

**Comment by Chidera6** [comment]
Hello guys. [Here](https://dev.to/chidera/setting-up-wagtail-for-development-on-windows-a-step-by-step-guide-j7p) is a step-by-step guide I wrote on setting up wagtail for development on Windows OS. It might prove helpful.

**Reply by Temiwaji** [reply]
Welldone and thank you @Chidera6

**Reply by Chidera6** [reply]
Thank you @Temiwaji

**Comment by waju-ire** [comment]
> Write a short blog post describing things you’ve learned recently, and share it with us. The post must be in English, include at least one image, be less than 500 words, and score a Grade 7 or lower on https://hemingwayapp.com/. You can describe any recent learnings – share a list of the last few articles you read and liked or focus on one specific thing you’ve been learning. It can be a broad area or a very narrow subject. It doesn’t have to be about tech. The post has to be posted on a publicly available platform (for example Dev.to, Hashnode, Medium, your own website, anywhere public), and you must also add it as a new section in your personal file in this repository (so we can provide feedback on the contents).

Hey guys, in our brief meeting with @thibaudcloud, he explained a bit on how the blog post grades work. So when you put your write ups on hemingwayapp.com, the lower the grade, the better your blog post gets. The screenshot👆🏽will provide better context.

Also, make sure to follow the instructions as laid e.g Have your word limit as instructed
Good luck with your tasks 🫡.

**Reply by MugoBrian** [reply]
Thank you.

**Reply by codewithmide** [reply]
That makes sense. Thanks for sharing

**Reply by Chidera6** [reply]
Thank you for this. Let me modify my post. I really thought a higher score was better.

**Reply by thibaudcolas** [reply]
Yes! It’s a "school" grade, so the idea with those readability tests is that you want your writing to be understandable by younger people. The "grade 7" we’re aiming for is quite low, so that this presents a bit of a challenge.

**Reply by thibaudcolas** [reply]
@waju-ire please avoid screenshots of text if possible. It’s not good for people who rely on assistive technologies to read content on the web, as they won’t be able to resize the text or have their screen reader read it to them.

**Reply by waju-ire** [reply]
> @waju-ire please avoid screenshots of text if possible. It’s not good for people who rely on assistive technologies to read content on the web, as they won’t be able to resize the text or have their screen reader read it to them.

Oh yes yes, Corrections noted
😅

**Reply by praiztech** [reply]
Thanks for the update @waju-ire

**Comment by thibaudcolas** [comment]
Q: Any recommendations of resources on accessibility related to Wagtail?

**Reply by thibaudcolas** [reply]
Most definitely!

Start with our [Accessibility policy](https://wagtail.org/accessibility), and our [Accessibility considerations documentation](https://docs.wagtail.org/en/stable/advanced_topics/accessibility_considerations.html). This second page has links at the very bottom for more generalist resources we’ve really liked.

**Comment by Ranecodes** [comment]
Hi guys, asking this for clarity, for our first blog post which is supposed to be about sharing something we have learned recently. Is it okay to write something completely not about tech, or does the topic have to circle back to tech or/and accessibility?

**Reply by waju-ire** [reply]
Hello, from what I understand about the instructions and from the meetings yesterday, it doesn't have to necessarily be about tech. Although, the mentors would like to see how what you have learned so far can contribute to the project.  I hope this helps😃

**Reply by Ranecodes** [reply]
Okay great, got it. Thanks!

**Reply by thibaudcolas** [reply]
**Yes**, it can be about something that’s completely not tech. I think I’d actually encourage that! Most people write about tech, and picking something different (and likely more personal) will make you stand out a bit. There’ll be plenty of opportunities to show your tech skills with other tasks.

Above all else – don’t overthink it. The main thing we want is for you to prove you can write and follow instructions (word count, image with proper alt text, grade target, etc). Hopefully doing so in a way that’s enjoyable for you.

The instructions for ref: 

> Write a short blog post describing things you’ve learned recently, and share it with us. The post must be in English, include at least one image, be less than 500 words, and score a Grade 7 or lower on https://hemingwayapp.com/. You can describe any recent learnings – share a list of the last few articles you read and liked, or focus on one specific thing you’ve been learning. It can be a broad area or a very narrow subject. It doesn’t have to be about tech. The post has to be posted on a publicly-available platform (for example Dev.to, Hashnode, Medium, your own website, anywhere public), and you must also add it as a new section in your personal file in this repository (so we can provide feedback on the contents).

**Reply by Ranecodes** [reply]
Thank you very much @thibaudcolas!

**Comment by ghost** [comment]
Hello everyone, i need help 😢

As suggested by @waju-ire to talk about issues we might face out of the project here, i am seeking help on few issues related to git. 

Being called a community, i believe i can ask for help here, i am trying to create a new branch and file in there for the task we were being provided of adding README.md file in 2023 file, but i am not able to commit the changes as i am always prompted by an error: 'File couldn't be edited'. I am new to git and github hence, went through few blunders already and don't wanna waste any more time so it would be really kind if i get some guidance as i really wanna start contributing to the main issues. 
Thank you and looking forward to get any help, asap. 😢

**Reply by Ranecodes** [reply]
Hi @Sushifyyy, I think that it's great you reached out, that's the essence of this community. We are all here to help one another.

Concerning your issue with committing changes, it's possible that you are creating a file with the same name as a file that already has that name in the repository.

First things first, make sure that you are creating a folder (and not a file) in the 2023 folder. Here is a step-by-step instruction on how to make a folder and add files inside it: https://intellipaat.com/community/20249/create-folder-in-github-how-to-create-folder-in-github-repository

For more clarity, when you click on Add new file: It should look like this outreachy/2023/YourName/README.md

I hope this helps.
Let me know if you have any more questions.

**Reply by ghost** [reply]
@Ranecodes thank you for the quick response, i appreciate it much. Unfortunately, it did not work out this time as well. And i am stuck at the same issue again😢
 Is there any way i can get connect to you for help regarding this issue?

**Reply by Chinex-Boroja** [reply]
@Sushifyyy are you creating the branch and working on the file from the GitHub website or you are using an IDE?

Secondly, I hope you forked and cloned the Wagtail outreachy repository to get your own copy.

**Reply by ghost** [reply]
> @Sushifyyy are you creating the branch and working on the file from the GitHub website or you are using an IDEI


I am using Github GUI.

**Reply by Chinex-Boroja** [reply]
In most cases, it could be you are creating a file that has the same name as the one you are trying to create. Check this [thread](https://github.com/orgs/community/discussions/54940#:~:text=Solution%20in%20my,was%20a%20mistake.), it might provide more context.

Let me know if it helps @Sushifyyy

**Reply by kofodym** [reply]
@Sushifyyy sorry about this errors. They can be a headache sometimes. 
1. Firstly let's know the branch you are on using the command git branch. 
2. What's your file structure? Is it Outreachy> your_folder_name, then your are trying to create a README.md in here right?
3. You said " i am trying to create a new branch and file in there for the task we were being provided of adding README.md file in 2023 file, ". You are not to add a README.md in 2023 folder instead you are to  "add a new folder with your name as the folder name" inside 2023/"  then inside that folder add a README.md

**Reply by ghost** [reply]
thank you @kofodym for responding. As from the point 3 of yours, i guess i am not doing anything wrong. In the 2023 folder of the wagtail/outreachy repo, i am creating a folder with my name in the 2023 folder like this:  outreachy/2023/<my_name>/README.md  .

**Reply by ghost** [reply]
i have forked the repo and went on creating a file by going to  'Add file>create new file', writing  2023/<my_name>/README.md in the 'Name Your file field' and copying my intro in the 'file contents zone'. The moment i proceed on clicking the 'commit changes' button, i am getting this error:  "there was an error commiting your changes: file couldn't be edited", hence not able to commit.

As a backdrop incident, i want to mention that i am trying to do all of the above in a separate branch now, as 2 days ago, i did a blunder at first while doing the task of forking and adding README.md and ended up commiting it accidentally. Somehow i reverted it but it still shows in my commit history. 
please tell me if i am doing something wrong again?

**Reply by EstherChristopher** [reply]
Hello @sushifyyy, do you mind we hop on a Google meet call so you can share your screen?

**Reply by ghost** [reply]
> Hello @sushifyyy, do you mind we hop on a Google meet call so you can share your screen? 

Thank you @EstherChristopher, yes sure, would you mind connecting on Twitter or any other platform?

**Reply by ghost** [reply]
> > Hello @sushifyyy, do you mind we hop on a Google meet call so you can share your screen? 
> 
> Thank you @EstherChristopher, yes sure, would you mind connecting on Twitter or any other platform?

I have sent you text on Twitter DM, mind checking it please?

**Reply by EstherChristopher** [reply]
> > > Hello @sushifyyy, do you mind we hop on a Google meet call so you can share your screen? 
> > 
> > Thank you @EstherChristopher, yes sure, would you mind connecting on Twitter or any other platform?
> 
> I have sent you text on Twitter DM, mind checking it please?

I've replied to you.

**Reply by thibaudcolas** [reply]
@Sushifyyy I think it’s good for you to reach out for help – but with my Outreachy coordinator hat on, I have to flag that knowledge of git and GitHub are one of our [expectations for contributors](https://github.com/wagtail/outreachy/blob/main/contributor-guide.md#contributor-expectations):

> No matter which specific project idea you’re interested in, we expect all participants to demonstrate:
> […]
> Knowledge of git and GitHub: coders or not, those are fundamental tools for how we work as a community.

Now there is time to catch up but I would suggest you go through more fundamental training with those tools before you spend further time going through our Outreachy / Wagtail - specific tasks.

**Reply by ghost** [reply]
> @Sushifyyy I think it’s good for you to reach out for help – but with my Outreachy coordinator hat on, I have to flag that knowledge of git and GitHub are one of our [expectations for contributors](https://github.com/wagtail/outreachy/blob/main/contributor-guide.md#contributor-expectations):
> 
> > No matter which specific project idea you’re interested in, we expect all participants to demonstrate:
> > […]
> > Knowledge of git and GitHub: coders or not, those are fundamental tools for how we work as a community.
> 
> Now there is time to catch up but I would suggest you go through more fundamental training with those tools before you spend further time going through our Outreachy / Wagtail - specific tasks.




Thank you @thibaudcolas for the nudge :) Fortunately, I was able to get out of the problem I ran into. And I do acknowledge your advice of going through fundamental training before proceeding with Wagtail tasks. I'm on it with a positive energy.

**Reply by ghost** [reply]
> > > > Hello @sushifyyy, do you mind we hop on a Google meet call so you can share your screen? 
> > > 
> > > Thank you @EstherChristopher, yes sure, would you mind connecting on Twitter or any other platform?
> > 
> > I have sent you text on Twitter DM, mind checking it please?
> 
> I've replied to you. 

Thank you @EstherChristopher :)  guess what, i was able to figure out the solution just in time.

**Reply by ghost** [reply]
> > > > Hello @sushifyyy, do you mind we hop on a Google meet call so you can share your screen? 
> > > 
> > > Thank you @EstherChristopher, yes sure, would you mind connecting on Twitter or any other platform?
> > 
> > I have sent you text on Twitter DM, mind checking it please?
> 
> I've replied to you. 

Thank you, guess what, i was able to figure out the issue and mend everything on time.

**Comment by EstherChristopher** [comment]
Hey guys👋🏻  

### You can use this software in order to make sure your blog posts are free from any grammatical errors or sentence misconstruction:  

-  Grammarly (To check your spelling, grammar, and English mistakes)

- Quillbot (A software similar to Grammarly)

- [Plagiarismdetection.net](http://plagiarismdetection.net/) (A website which checks for plagiarism in your article)

### You can also take note of the following to ensure the delivery and structuring of your blog post: 

-  Prioritize active over passive voice, eg. The software was installed by the technician.❌ The technician installed the software ✅

-  Use simple and easy-to-read language to improve clarity and prevent ambiguity  

- Avoid too lengthy sentences. Break down sentences for readability

-  Be concise, remove verbose explanations. Especially considering the word length we are to work with, so the blog posts can capture important details

-  Clear conclusion to sum up everything written in the blog post

-  Make use of headings and subheadings  

-  Use a catchy title (You could score more points for this)

-  Do not forget to include an image with alt text (Accessibility hint😜)  

- Make sure your main point stands out either through formatting or emphasizing it in the conclusion

-  Make your intro engaging to capture the reader’s attention. The intro is an important part of a blog post.  

-  You might also want to write in Google Docs first before copying and pasting can help you keep track of your word count. 

You can also include any other software you know of that makes your articles better✨

**Reply by Ranecodes** [reply]
This is very helpful Thanks for this @EstherChristopher

**Reply by EstherChristopher** [reply]
You're welcome @Ranecodes

**Reply by thibaudcolas** [reply]
> Do not forget to include an image with alt text (Accessibility hint😜)  

😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😉 😬

**Reply by kofodym** [reply]
Nice one.

**Reply by kofodym** [reply]
May I add these small details.
- A captivating image: It should be renamed to what it's about e.g. an image with a cat could be cat.jpeg instead of image2.jpeg. Incase the user has a slow connection, this Image name helps with  readability.

- Also a good heading: It should contain keywords associated with what you are writing about. 
It helps improve your search ranking and also helps those who can't see have an idea of what to expect in the content.

**Reply by EstherChristopher** [reply]
Great @kofodym 👍🏻

**Reply by MugoBrian** [reply]
Hello @EstherChristopher.
Thank you for the resources, however there's correction I would love to suggest. I believe its https://plagiarismdetector.net/ instead of https://plagiarismdetection.net.

**Reply by MugoBrian** [reply]
Hello everyone. Are there alternatives to the one provided[ above](https://plagiarismdetector.net/ ). The tool is quite slow.

**Comment by triumphant-fortune** [comment]
Greetings everyone 👋

### Writing and Editing Tips for Blog Posts with Hemingway Editor

**Start Writing First:**

Begin by writing your blog post without worrying about grammar and style. Get your ideas on paper first.

**Use Hemingway Editor**:

Copy and paste your written content into Hemingway Editor, including sections or the entire post.

**Check Readability:**

Pay attention to the suggested readability level in Hemingway. Aim for a level that suits your audience. Lower levels are often better for online content.

**Avoid Adverbs and Passive Voice:**

Hemingway Editor points out adverbs and passive voice. Review and consider replacing them with stronger verbs.

**Simplify Complex Sentences**:

Complex sentences can be hard to read. Break them into shorter, more understandable sentences when possible.

 **Use Clear and Simple Language**:

Hemingway Editor highlights complex words and phrases. Simplify your language for better reader understanding.

  **Mind Sentence Length**:

Hemingway Editor identifies overly long sentences. Shorter sentences are usually easier to read and understand.

**Vary Word Usage**:

Hemingway Editor spots repeated words. Diversify your word choice to make your writing more engaging.

**Organize Paragraphs**:

Follow Hemingway's suggestions to improve paragraph structure, ensuring each one has a clear focus.

**Check Readability Issues**:

 Hemingway Editor marks sentences with readability problems. Review and rewrite them for clarity.

 **Maintain Natural Flow**:

 While following Hemingway's advice is crucial, keep your writing natural. Avoid making it sound robotic.

**Review After Edits**:

After making recommended edits, reread your blog post to ensure it still conveys your intended message and tone.

  **Grammar and Spelling**:

Hemingway Editor focuses on style and readability, so use a separate grammar and spell-check tool or proofread manually for errors.

**Practice Consistently**:

Regularly using Hemingway Editor helps you develop good writing habits, making your blog posts more readable and engaging.

You can also learn more about Plain Language Web Writing Tips by visiting this helpful link: https://digital.gov/resources/plain-language-web-writing-tips/. It provides valuable information and advice for writing clearly on the web.

Feel free I am open to suggestions, corrections, and contributions do all have a nice day ahead🤗

**Reply by Chinex-Boroja** [reply]
Thank you @triumphant-fortune

**Reply by kofodym** [reply]
Well done. Thank you

**Comment by nouman17610** [comment]
@waju-ire 
I want to ask about contributions. Should we add contributions in Outreachy after every task or should we add the main final task like fixing the issues as a "record contribution"?
Right now I am writing my first blog.
Thank you

**Reply by Chinex-Boroja** [reply]
Probably @thibaudcolas can answer this

**Reply by EstherChristopher** [reply]
Can you better reword this question?

**Reply by nouman17610** [reply]
@EstherChristopher Does every task on the contributor guide require a contribution recording in Outreachy?

**Reply by waju-ire** [reply]
It's better you add after every task. I think that's how it should be.

**Reply by EstherChristopher** [reply]
I can't say for sure. Because some of those activities are not contributions in themselves. For example, creating a GitHub account or introducing yourself on the discussion group or Slack channel. 

But then this question will best be answered by Thibaud himself so let's wait for him.

**Reply by thibaudcolas** [reply]
You asked a slightly different question elsewhere so I’ve answered that one first, which might help. See https://github.com/wagtail/outreachy/discussions/30#discussioncomment-7273016.

As far as _what_ to record as a contribution – definitely _not_ every task.

Take a broad definition of what it means to contribute to a project, but keep a tight focus on quality. Read [How to contribute](https://opensource.guide/how-to-contribute/) from Open Source Guides. My take:

- Contributions can take lots of different shapes. Avoid the narrow definition of "it only counts if it’s code that’s committed". Helping others, writing about the project/community to your audience, reporting issues, asking good questions others can learn from – those are all great ways to contribute.
- For Outreachy specifically, the point of recording contributions is to bolster your application. Think of which contributions mentors will convince mentors you have what it takes to take on the internship on their project.

**Comment by nouman17610** [comment]
I have another question. 
Do I need to create a branch when I want to create a new pull request? I fork the repository and create a new branch for every pull request. Is it the right way?

**Reply by Chinex-Boroja** [reply]
Hi @nouman17610 you really don't need to. But since you have created a branch, you are still on the right path. Once you are done, you can raise your pull request

**Reply by nouman17610** [reply]
@Chinex-Boroja any idea about pull request merge? My pull requests still require a merge from mentors

**Reply by Chinex-Boroja** [reply]
You don't need to do anything again. The mentors will be the ones to merge your PR, once you have sent it. You just need to exercise a bit more patience. They will review it and give you a feedback if need be.  

Just wait a bit

**Reply by nouman17610** [reply]
Okay got it.  I will continue the process and all the PRs will be reviewed by Mentors at the end.
Thank you

**Reply by Chinex-Boroja** [reply]
Yes, exactly. 
You are welcome

**Reply by thibaudcolas** [reply]
@nouman17610 yup, @Chinex-Boroja provides solid guidance here. I’d add it’s normal practice to create branches for all pull requests, as it really helps when you work on multiple things at the same time.

**Reply by kofodym** [reply]
@thibaudcolas great. Creating branches makes sense.
I created just one branch and I noticed that my commits are added to the unmerged PR. Making that one PR to contain multiple tasks. Thank you

**Reply by nouman17610** [reply]
@thibaudcolas Yes I have been creating separate branches for each Pull Request. I wasn't sure that it would be the right way. Thank you for clarifying the issue.

**Reply by maureenblack** [reply]
Hello everyone, I need help. I have a little trouble understanding what exactly I am supposed to do with the articles/blog posts. I know i need to publish on a public channel, but i am confused about the part where it says "and you must also add it as a new section in your personal file in this repository (so we can provide feedback on the contents)." Heeelp!

**Reply by nouman17610** [reply]
@maureenblack you just need add separate section and provide link of the article under that section. Section is actually a heading of blog

**Reply by kofodym** [reply]
@maureenblack Add a new section using ##Section-name inside your readme.md file

**Reply by maureenblack** [reply]
Thank you both very much 🙏🏽

**Comment by thibaudcolas** [comment]
Q from @nouman17610: [When would you like us to record contributions at outreach? with every task or only the main issue?](https://github.com/wagtail/outreachy/discussions/3#discussioncomment-7261165)

**Reply by thibaudcolas** [reply]
A: Good question. To be honest I’m not sure, as the contribution recording process on the Outreachy site isn’t described much for coordinators like me. The only info I have is the [Recording your contributions](https://www.outreachy.org/docs/applicant/#record-contributions) docs in the Applicant Guide, and what I’ve seen of how this works from last year.

My understanding is that:

- It doesn’t matter _when_ contributions are recorded, except for the first one (see below). From the mentors’ side, we only see contributions at the very end of this application period (so late October) when people actually submit an application.
- The very first contribution is useful to record early in case we were to "close" the project to new applicants. This is something Outreachy coordinators can do if they believe they already have enough interest. I don’t see ourselves doing that, but it’s a possibility. If it did happen, people who have already recorded one contribution will be able to record more. Even if we did do that – I’d probably give you all a heads’up, and also if you contacted me afterwards because of issues, reopen the project to sort things out. So not the biggest deal but worth being aware of.

As far as how granular to record contributions – I would recommend focusing on quality over quantity. The main point of recording contributions is to convince mentors your application is solid.

So – from my side, I’d say either record them as you go if you think that works for you, or only record one because of that "closed project" rule, and then towards the end of the application period you can review what you’ve done more critically and decide what to record as a contribution to tell a more cohesive story.

**Reply by kofodym** [reply]
Thank you for the explanation.

**Reply by EstherChristopher** [reply]
Thank you for shedding light on this.

**Reply by Chinex-Boroja** [reply]
Thank you for this

**Reply by nouman17610** [reply]
@thibaudcolas thank you for the in-depth explanation.

**Reply by triumphant-fortune** [reply]
Thank you for the insights @thibaudcolas

**Comment by oly-baby** [comment]
Thank you so much

**Comment by nouman17610** [comment]
Another important question.
when I create a new branch from the existing branch and create a pull request, a new pull request does not appear in the Outreachy pull requests. New pull requests are appearing on my profile like nouman17610/outreachy. It is fine?
@thibaudcolas please you can respond.

**Reply by thibaudcolas** [reply]
@nouman17610 no, pull requests need to be created on https://github.com/wagtail/outreachy. Please avoid `@-mention`-ing people unless it’s really critical – lots of other people could answer this just as well as I might.

**Reply by codewithmide** [reply]
Hi @nouman17610 You can submit multiple PRs. Whenever you are done making new changes to your branch, just sync the fork and submit a new PR. You got this
![sync](https://github.com/wagtail/outreachy/assets/81376427/5209648c-5a43-4a06-8b7a-e6b8b28ae753)

**Reply by nouman17610** [reply]
@codewithmide thank you

**Comment by EstherChristopher** [comment]
Hey guys👋🏻
I posted this on our Slack channel and I thought to post this here too.

Did some digging around and I thought to share some baseline knowledge I found about Wagtail.
It should help you get familiar with Wagtail✨

- Wagtail is a content management system (A CMS is a software application used to create and manage digital content)

- It is open-source and based on the Django framework which was written in Python.

- Wagtail is a traditional CMS with some support for headless use (In traditional, the content management backend and frontend live on a single server. In headless, the content management backend is separate from the frontend).

- Wagtail has a user-friendly admin interface which makes it easy for content editors to use.

- Wagtail aims to document all of its accessibility features in the CMS in line with the [ATAG 2.0](https://www.w3.org/TR/ATAG20/) standard and hence the need for the Accessibility features documentation project.

- Wagtail aims to improve its website to address accessibility issues and possible improvements in its audit hence the need for the [Wagtail.org](http://wagtail.org/) website accessibility project.

- Wagtail CMS supports multiple languages.

- Wagtail has capabilities to be used as a headless CMS but it is still implementing some features for Wagtail’s developers.

- Wagtail has an active and supportive community with regular updates to the core system, such as the projects on Outreachy.

- Last tip😉: Popular organizations, including NASA and Google, have used Wagtail for their projects.

I’m open to corrections from our mentor in case of any potentially factually inaccurate statement🤗

**Reply by oly-baby** [reply]
Thank you for this Esther. This is so helpful

**Reply by EstherChristopher** [reply]
You're welcome.

**Reply by thibaudcolas** [reply]
@EstherChristopher I think this is helpful but also the kind of information I would expect people to do their own research for? Focus on the tidbits of this that are not readily-available information. As-is, it looks like you asked an AI assistant to do your research and just pasted the output here.

**Reply by EstherChristopher** [reply]
Oh, I didn't see this soon enough. 
Alright, thanks. I thought it was helpful to share. 
Ah, no. I didn't ask an AI assistant to generate this. I read about Wagtail on several of its sites and curated what I gathered. 
Thanks for the feedback.

**Comment by Ranecodes** [comment]
## Hey everyone! 👋 I stumbled upon this fantastic resource and thought of sharing key insights about GitHub pull requests that we can all benefit from. 🚀

These are 10 valuable tips on how to make the most out of GitHub pull requests for our projects! 🌟

1. **Write a Clear and Concise Description**: Clearly describe the changes made in the pull request to help reviewers understand the purpose and context.

2. **Keep Commits Small**: Small, focused commits simplify the review process and make it easier to revert code if needed.

3. **Test Code Before Sending**: Always ensure that your code is thoroughly tested and functional before creating a pull request.

4. **Use Branch Protection Rules**: Set up branch protection rules to maintain code quality and prevent accidental merges.

5. **Follow Coding Standards**: Stick to coding standards and conventions used in the project to maintain consistency.

6. **Avoid Self-Merging**: Allow other team members to review and merge your pull requests to ensure a second set of eyes on the changes.

7. **Rebase Regularly**: Keep your branch up to date with the main repository by regularly rebasing it to the latest changes.

8. **Request Feedback**: Ask for feedback from team members and incorporate their suggestions to improve your code.

9. **Keep Pull Requests Small and Focused**: Focus on specific changes in each pull request to make reviews more manageable and effective.

10. **Monitor and Maintain Pull Requests**: Stay engaged with the pull request discussion, address comments, and keep the conversation going until the changes are merged.

For more in-depth information and tips about GitHub pull requests, check out [this article](https://blog.mergify.com/github-pull-requests-10-tips-to-know/).

Source: [GitHub Pull Requests: 10 Tips To Know](https://blog.mergify.com/github-pull-requests-10-tips-to-know/)

**Reply by Chinex-Boroja** [reply]
Insightful. Thank you @Ranecodes

**Comment by MugoBrian** [comment]
Greetings, all! While on my journey to grasp the concept of web accessibility, I've acquired insights into assistive technologies. I'm eager to offer some valuable resources that can aid in comprehending ARIA (Accessible Rich Internet Applications), a fundamental set of guidelines and practices to guarantee smooth website navigation for users employing assistive technologies. For a comprehensive collection of web accessibility (often denoted as a11y) materials, please feel free to explore the following:

[Explore ARIA - Accessible Rich Internet Applications](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA).
[Understand Assistive Technologies](https://www.w3.org/WAI/people-use-web/tools-techniques/)

For a broader array of web accessibility resources and tools, please visit:
1.  [W3C's Assistive Technologies page](https://www.w3.org/WAI/people-use-web/tools-techniques/).
2. [Mozilla Web Accessibility Docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility)

**Reply by oly-baby** [reply]
Thank you so much Brian for this resources

**Reply by thibaudcolas** [reply]
Nice one @MugoBrian, those are great resources.

**Reply by cedarworth** [reply]
Thanks @MugoBrian for the resources

**Reply by EstherChristopher** [reply]
Thank you @MugoBrian This will streamline our research

**Comment by nishantn31** [comment]
Hi @thibaudcolas I have raised a PR on the wagtail/wagtail.org issue and one in the wagtail/outreachy (README.md update).
Kindly guide me which of the PR I should record as contribution or I can mention both?

**Reply by nouman17610** [reply]
@nishantn31 there is no specific guideline about how to record contribution for various PRs. You can record contribution for PRs that contain important content. 
So you can raise for both.

**Reply by thibaudcolas** [reply]
See https://github.com/wagtail/outreachy/discussions/30#discussioncomment-7273016.

**Comment by praiztech** [comment]
For anyone still trying to grasp the concept of web accessibility, check out [WebAIM's Introduction to Accessibility](https://webaim.org/intro/) post.

If you'd prefer short and easy guides on how to make web components accessible, check out the [WAI tutorials](https://www.w3.org/WAI/tutorials/page-structure/).

**Comment by waju-ire** [comment]
Hey guys, if you're struggling with communicating your ideas on the blogposts, check this podcast:

[Think Fast, Talk Smart: The Podcast](https://www.gsb.stanford.edu/business-podcasts/think-fast-talk-smart-podcast)
They talk about great methods of communication, from writing documentations to public speeches.  
Their episodes are quite short and easy to understand. Keep up with the contribution👏🏽👏🏽

**Reply by thibaudcolas** [reply]
Please make sure to add link text to your links. It’s not super clear what the podcast is currently. People who use assistive tech can find it hard to navigate links as URLs.

**Reply by waju-ire** [reply]
Thanks for the heads up 😅.

**Reply by maureenblack** [reply]
My biggest challenge with the blog post is making it to grade 7 in the hermingwayapp. the best i could do was 9, and i dont know what to do anymore.

**Reply by waju-ire** [reply]
Hi Maureen, one of the things I do is construct simple short sentences  e.g rather than say " the problem has many layers to it which will help in defining the research framework" say "The problem has many layers to it. This will help to define the research framework".  Hemingwayapp marks sentences with lots of conjunctions as hard to read.
Also, you want to personalise your post, like describe an experience you had and put it in context with the topic.

I hope this helps; we can text more on slack if you need more clarification😅

**Reply by EstherChristopher** [reply]
> My biggest challenge with the blog post is making it to grade 7 in the hermingwayapp. the best i could do was 9, and i dont know what to do anymore.

It took me a while too but I eventually got it to a grade 7. One thing that helped me was asking Google. For instance, I'd ask "what are other easier words for "Conversely", or how do I change "is displayed" to active voice. Then I'd pick the word suggested that mostly fits into the sentence. Doing this worked for me.

I hope this tip helps.
